lesson17/src/js/main.js

```diff
@@ -0,0 +1,212 @@
+
+  const nextButton = document.createElement("button");
+  nextButton.classList.add("arrow", "next");
+  nextButton.id = "js-button_next";
+  nextButton.textContent = "next";
+  nextButton.setAttribute("aria-label", "next");
+
+  fragment.appendChild(nextButton);
+
+  const slideArea = document.querySelector(".mainvisual__images");
+  slideArea.appendChild(buttonArea).appendChild(fragment);
+}
+
+function renderPagenation(item) {
+  const pagenation = document.createElement("div");
+  pagenation.classList.add("slider__pagination");
+
+  const fragment = document.createDocumentFragment();
+
+  const current = document.createElement("span");
+  current.id = 'js-current';
+  current.textContent = 1;
+  fragment.appendChild(current);
+
+  const separation = document.createElement("span");
+  separation.textContent = "/";
+  separation.classList.add("separation");
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.id = 'js-total';
+  total.textContent = item.length;
+  fragment.appendChild(total);
+
+  const textArea = document.querySelector(".mainvisual__textarea");
+  textArea.appendChild(pagenation).appendChild(fragment);
+}
+
+function manipulatePagination(target) {
+  const pagenationCurrent = document.getElementById("js-current");
+  const pagenationTotal = Number(document.getElementById("js-total").textContent);
+  let currentNum = Number(pagenationCurrent.textContent);
+
+  const buttonPrev = target.getAttribute("aria-label") === 'previous';
+  const buttonNext = target.getAttribute("aria-label") === 'next';
+
+  if (currentNum < pagenationTotal && buttonNext) {
+    pagenationCurrent.textContent = currentNum +=1;
+  }else if(currentNum <= pagenationTotal && buttonPrev){
+    pagenationCurrent.textContent = currentNum -= 1;
+  }
+}
+
+function slideImageNext(e) {
+  const slideImages = [...document.getElementsByClassName("slider__item")];
+
+  const activeSlide = document.querySelector(".is-active");
+  const nextSlide = activeSlide.nextElementSibling;
+  const nextSlideIndex = Number(nextSlide.getAttribute("data-slide-index"));
+  if (nextSlideIndex === slideImages.length) {
+    nextSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+    e.disabled = true;
+  } else {
+    e.previousElementSibling.disabled = false;
+    nextSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+  }
+}
+
+function slideImagePrev(e) {
+  const activeSlide = document.querySelector(".is-active");
+  const prevSlide = activeSlide.previousElementSibling;
+  const prevSlideIndex = Number(prevSlide.getAttribute("data-slide-index"));
+
+  if (prevSlideIndex === 1) {
+    prevSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+    e.disabled = true;
+  } else {
+    e.nextElementSibling.disabled = false;
+    prevSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+  }
+}
```

Comment:
I thought I could summarize the function.
I also thought it would be easier to understand if the functions that turn on and off disabled were separated out.

```suggestion
function switchImg(direct) {
  const active = document.querySelector(".is-active");
  active.classList.remove("is-active");
  active[direct].classList.add("is-active");
}

function toggleButtonDisabled() {
  const slideImages = [...document.getElementsByClassName("slider__item")];
  const activeImg = document.querySelector(".is-active");
  const currentIndex = Number(activeImg.dataset.index);
  const firstIndex = 0;
  const lastIndex = slideImages.length - 1;
  prevButton.disabled = currentIndex === firstIndex;
  nextButton.disabled = currentIndex === lastIndex;
}

```
When using switchImg, it is possible to slide back and forth by using 'nextElementSibling' and 'previousElementSibling' as arguments as follows.

```
  switchImg("nextElementSibling");
  switchImg("previousElementSibling");
```

Also, I believe the following statement in init() can be removed by executing toggleButtonDisabled().
```
  const prevButton = document.querySelector('[aria-label="previous"]');
  prevButton.disabled = true;
```

Reply:
Thank you for your review!
Conditional ternary operator I learned a lot.

part of switcImg Because I didn't have this idea at all
So that's right！！！！

[087a902](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/087a90223ca71d79cd85d32bcda7e60efe3b780e)

Reply:
nits
Thank you for your fixed!
The 'switch' part is typo.

Reply:
Thank you for your review!
Fixed that h of switch was missing.

[b54e646](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/b54e646f9898fa655d01183659135fcc3dab1b96)